common/djangoapps/third_party_auth/saml.py

```diff
@@ -231,6 +231,11 @@ def odata_company_id(self):
     def odata_client_id(self):
         return self.conf['odata_client_id']
 
+    @property
+    def odata_user_request_fields(self):
+        """ Optional property for overriding OData User field specification """
+        return self.conf.get('odata_user_request_fields')
```

Comment:
Why not just set the `sapsf_field_mappings` config value instead?

Reply:
From what I was able to grok in the code + see in the test debugger, it looks like `sapsf_field_mappings` simply override the existing 'default_field_mappings' -- the end result is that the system still makes requests which include the "city" field.

Reply:
Maybe we should modify the behavior of `sapsf_field_mappings` then instead of introducing another configuration value.

Reply:
Hmm, ok -- seems like a lot more work -- do you have a recommendation for how you'd prefer to do that?

Reply:
Seems like you could modify the `field_mappings` property pretty easily to completely override `default_field_mapping` instead of just updating it.